thorlcr/activities/funnel/thfunnelslave.cpp

```diff
@@ -141,14 +142,15 @@ class CParallelFunnel : implements IRowStream, public CSimpleInterface
     Linked<IOutputRowSerializer> serializer;
 
     void push(const void *row)
-    {   
+    {
         size32_t rowSize = thorRowMemoryFootprint(serializer, row);
 
         bool waitForSpace = false;
         // only allow a single writer at a time, so only a single thread is waiting on the semaphore - otherwise signal() takes a very long time
         {
-
+            BlockedActivityTimer timer(activity.getActivityTimerAccumulator(), activity.queryTimeActivities());
```

Comment:
why measure this blocking time (on crit), but not on waitSem blockage?

Reply:
I am not sure this time should be considered blocked at all.

As it stands, if there are 3 inputs to the funnel, the 1st will gain this crit instantly, and then (if full) hold the crit and wait on the fullSem semaphore (but will not count toward blocked time)
The other 2 input handlers will both block here and count toward blocked time.
If there are 100 inputs, 99 will add blocked time.

But I'm not sure any should consider this blocked time, they are not "blocking" the downstream act *except* for the very small amount of time they spend in the crit adding to 'rows'.
If they (the input handlers) are blocked on fullSem it's because the downstream act isn't pulling or pulling fast enough - this act. is not blocking it.